### PR TITLE
Share on Audio.com update (#19115)

### PR DESCRIPTION
**Pull request by mathesoncalum**
Resolves: #19115

TODOs:
- [x] Update the dialog (allow the user to choose whether to replace the existing file)
- [x] Save the existing file id in "File -> Project properties" (preserve between sessions)
- [x] Use PATCH /audio/{id} to update the visibility of an existing file
- [x] Refactor `visibilityDropdownModel` function in `SaveToCloudDialog.qml`
- [x] Update "File->Project Properties" to store the Audio.com URL (instead of the raw ID)

### CODE REVIEW

File: src/cloud/view/cloudsmodel.cpp [hunk 1]
```diff
@@ -141,6 +146,63 @@ QVariant CloudsModel::cloudInfo(const QString& cloudCode) const
...
+    publicVisibility.insert("value", int(Visibility::Public));
+    publicVisibility.insert("text", qtrc("project/save", "Public"));
+    visibilityTypes.append(publicVisibility);
+
+    QVariantMap unlistedVisibility;
+    unlistedVisibility.insert("value", int(Visibility::Unlisted));
+    unlistedVisibility.insert("text", qtrc("project/save", "Unlisted"));
+    visibilityTypes.append(unlistedVisibility);
+
+    if (cloudCode == "musescorecom") {
```
Comment (Eism) [inline src/cloud/view/cloudsmodel.cpp:163]: cloud::MUSESCORE_COM_CLOUD_CODE

File: src/project/qml/MuseScore/Project/SaveToCloudDialog.qml [hunk 2]
```diff
@@ -145,11 +150,7 @@ StyledDialogView {
                     StyledDropdown {
                         Layout.fillWidth: true
 
-                        model: [
-                            { value: CloudVisibility.Public, text: qsTrc("project/cloud", "Public") },
```
Comment (Eism) [inline src/project/qml/MuseScore/Project/SaveToCloudDialog.qml:149]: QMLCloudVisibility is no longer needed

**Comment by zacjansheski** [comment]
Tested on Mac OS 11.6.5, Ubuntu 20.04.6, Windows 11, approved.
https://github.com/musescore/MuseScore/issues/19115 FIXED

**Review by Eism — APPROVED** [review]

**Comment by zacjansheski** [comment]
If we are in agreeance:

We can merge this 
Tested on Mac OS 11.6.5, Ubuntu 20.04.6, Windows 11.  Approved
https://github.com/musescore/MuseScore/issues/19115 FIXED

And I can log that issue once it is merged.

**Pull request merged**